deployability/modules/allocation/generic/models.py

```diff
@@ -95,5 +105,26 @@ def check_working_dir(cls, v: str | Path) -> Path:
         return path
 
 
-class DeletionPayload(InputPayload):
+class TrackPayload(BaseModel):
     track_output: Path
+
+class DeletionPayload(BaseModel):
+    instance_dir: Path
+    identifier: str
+    key_path: Path
+    platform: str
+    host_identifier: str | None = None
+    host_instance_dir: Path | None = None
+    ssh_port: str | None = None
+    arch: str
+
+class InstancePayload(BaseModel):
+    path: str | Path
+    identifier: str
+    platform: str
+    host_identifier: str  | None = None
+    host_instance_dir: str | Path | None = None
+    macos_host_parameters: dict | None = None
+    arch: str | None = None
+    ssh_port: str | None = None
+    user: str | None = None
```

Comment:
As this classes have some repeated attrs, and both are payloads, I think both could inherit from a `BasePayload` class that have defined those attributes, note that in the cases where one attr is required in one class but not in the other, it should be redefined overriding it. For example the attr `arch`.